src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1113 @@
+        IL_0006: ret
+    } // end of method Base::.ctor
+    // Properties
+    .property instance int32 Prop()
+    {
+        .custom instance void System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+            01 00 00 00
+        )
+        .get instance int32 Base::get_Prop()
+        .set instance void Base::set_Prop(int32)
+    }
+} // end of class Base
+");
+
+        verifier.VerifyTypeIL("Derived", @"
+.class private auto ansi beforefieldinit Derived
+    extends Base
+{
+    .custom instance void System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+        01 00 00 00
+    )
+    // Fields
+    .field private int32 '<Prop>k__BackingField'
+    .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+        01 00 00 00
+    )
+    // Methods
+    .method public hidebysig specialname virtual 
+        instance int32 get_Prop () cil managed 
+    {
+        .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+            01 00 00 00
+        )
+        // Method begins at RVA 0x2069
+        // Code size 7 (0x7)
+        .maxstack 8
+        IL_0000: ldarg.0
+        IL_0001: ldfld int32 Derived::'<Prop>k__BackingField'
+        IL_0006: ret
+    } // end of method Derived::get_Prop
+    .method public hidebysig specialname virtual 
+        instance void set_Prop (
+            int32 'value'
+        ) cil managed 
+    {
+        .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+            01 00 00 00
+        )
+        // Method begins at RVA 0x2071
+        // Code size 8 (0x8)
+        .maxstack 8
+        IL_0000: ldarg.0
+        IL_0001: ldarg.1
+        IL_0002: stfld int32 Derived::'<Prop>k__BackingField'
+        IL_0007: ret
+    } // end of method Derived::set_Prop
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        // Method begins at RVA 0x207a
+        // Code size 7 (0x7)
+        .maxstack 8
+        IL_0000: ldarg.0
+        IL_0001: call instance void Base::.ctor()
+        IL_0006: ret
+    } // end of method Derived::.ctor
+    // Properties
+    .property instance int32 Prop()
+    {
+        .custom instance void System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+            01 00 00 00
+        )
+        .get instance int32 Derived::get_Prop()
+        .set instance void Derived::set_Prop(int32)
+    }
+} // end of class Derived
+");
+    }
+
+    [Fact]
+    public void RequiredMemberAttributeEmitted_AddRequiredOnOverride()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+class Base
+{
+    public virtual int Prop { get; set; }
+}
+class Derived : Base
+{
+    public override required int Prop { get; set; }
+}
+class DerivedDerived : Derived
+{
+    public override required int Prop { get; set; }
```

Comment:
Consider testing a scenario similar to this where 'DerivedDerived.Prop' is not required.